be/src/exec/spill/spill_components.cpp

```diff
@@ -0,0 +1,352 @@
+
+#include "exec/spill/serde.h"
+#include "exec/spill/spiller.h"
+#include "exec/spill/spiller.hpp"
+
+namespace starrocks::spill {
+// implements for SpillerWriter
+Status SpillerWriter::_decrease_running_flush_tasks() {
+    if (_running_flush_tasks.fetch_sub(1) == 1) {
+        if (_flush_all_callback) {
+            RETURN_IF_ERROR(_flush_all_callback());
+        }
+    }
+    return Status::OK();
+}
+
+const SpilledOptions& SpillerWriter::options() {
+    return _spiller->options();
+}
+
+bool RawSpillerWriter::has_pending_data() {
+    std::lock_guard guard(_mutex);
+    return _mem_table_pool.size() != options().mem_table_pool_size;
+}
+
+Status RawSpillerWriter::prepare(RuntimeState* state) {
+    if (!_mem_table_pool.empty()) {
+        return Status::OK();
+    }
+
+    const auto& opts = options();
+    for (size_t i = 0; i < opts.mem_table_pool_size; ++i) {
+        if (opts.is_unordered) {
+            _mem_table_pool.push(std::make_unique<UnorderedMemTable>(state, opts.spill_file_size, _parent_tracker));
+        } else {
+            _mem_table_pool.push(std::make_unique<OrderedMemTable>(&opts.sort_exprs->lhs_ordering_expr_ctxs(),
+                                                                   opts.sort_desc, state, opts.spill_file_size,
+                                                                   _parent_tracker));
+        }
+    }
+
+    return Status::OK();
+}
+
+Status RawSpillerWriter::flush_task(RuntimeState* state, const MemTablePtr& mem_table) {
+    if (state->is_cancelled()) {
+        return Status::OK();
+    }
+
+    const auto& serde = _spiller->serde();
+    spill::AcquireBlockOptions opts;
+    opts.query_id = state->query_id();
+    opts.plan_node_id = options().plan_node_id;
+    opts.name = options().name;
+    ASSIGN_OR_RETURN(auto block, _spiller->block_manager()->acquire_block(opts));
+
+    // TODO: reuse io context
+    SerdeContext spill_ctx;
+    // DCHECK(writable != nullptr);
+    {
+        SCOPED_TIMER(_spiller->metrics().write_io_timer);
+        // flush all pending result to spilled files
+        size_t num_rows_flushed = 0;
+        RETURN_IF_ERROR(mem_table->flush([&](const auto& chunk) {
+            num_rows_flushed += chunk->num_rows();
+            RETURN_IF_ERROR(serde->serialize(spill_ctx, chunk, block));
+            return Status::OK();
+        }));
+        TRACE_SPILL_LOG << "spill flush rows:" << num_rows_flushed << ",spiller:" << this;
+    }
+
+    // be careful close method return a not ok status
+    // then release the pending memory
+    // flush
+    RETURN_IF_ERROR(block->flush());
+    _spiller->block_manager()->release_block(block);
+
+    {
+        std::lock_guard<std::mutex> l(_mutex);
+        _block_group.append(std::move(block));
+    }
+
+    return Status::OK();
+}
+
+Status RawSpillerWriter::acquire_stream(const SpillPartitionInfo* partition,
+                                        std::shared_ptr<SpillInputStream>* stream) {
```

Comment:
why we need this function? seems partition in params is not used.

Reply:
we will call SpillerWriter::acquire_stream() later.